Using section identifiers and performance improvements over sorting

richard-ramos opened the pull request:
- Use a single message list instead of a message list per channel
- load messages on channel switch.
- Use section identifiers as a way to identify if messages have been sent by the current user (no need to modify the model with `repeatMessageInfo`)

jrainville commented on ui/app/AppLayouts/Chat/ChatColumn/Message.qml:
```diff
@@ -34,7 +35,7 @@ Item {
         fillMode: Image.PreserveAspectFit
         source: identicon
-        visible: repeatMessageInfo && !isCurrentUser
+        visible: authorCurrentMsg != authorPrevMsg && !isCurrentUser
```
Duh, lol it makes a lot of sense now that you show it haha

richard-ramos commented:
I reverted the code for the single message list and added the commit to the desktop notes, to revisit it later, since using a single message list adds some lag.
